src/Types/Output/OutputType.hack

```diff
@@ -5,9 +5,23 @@ use namespace Slack\GraphQL;
 interface IOutputType {
     require extends BaseType;
 
-    public function unwrapType(): NamedOutputType;
+    public function unwrapType(): INamedOutputType;
 }
 
+interface IOutputTypeFor<TExpected, TResolved> extends IOutputType {
```

Comment:
What is the benefit of using this interface (and `IInputTypeFor`) as opposed to the traits directly? E.g., could the constructor on `FieldDefinition` just accept an object typed as `TOutputTypeFor<TRet, TResolved>`?

Reply:
Traits can't be used as type annotations :( That's why we need a matching interface for every trait.

Reply:
TIL